tests/components/LocationBias.test.tsx

```diff
@@ -0,0 +1,143 @@
+      setUserLocation: jest.fn(),
+    }
+  });
+  jest.spyOn(locationOperations, 'getUserLocation').mockResolvedValue(newGeoPosition);
+  jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+});
+
+it('renders the proper text (location DisplayName, method, and update btn)', () => {
+  render(<LocationBias />);
+  const expectedLocationName = mockedStateVaDevice.location.locationBias.displayName;
+  const locationNameElement = screen.getByText(expectedLocationName);
+  expect(locationNameElement).toBeDefined();
+
+  const expectedMethodMessage = '(based on your device) -';
+  const basedOnMethodElement = screen.getByText(expectedMethodMessage);
+  expect(basedOnMethodElement).toBeDefined();
+
+  const updateLocationButton = screen.getByText('Update your location');
+  expect(updateLocationButton).toBeDefined();
+});
+
+it('calls setUserLocation with coordinates returned by getUserLocation as params when update location clicked', async () => {
+  const actions = spyOnActions();
+  render(<LocationBias />);
+  clickUpdateLocation();
+
+  const expectedCoordinates = {
+    latitude: newGeoPosition.coords.latitude,
+    longitude: newGeoPosition.coords.longitude
+  };
+
+  expect(locationOperations.getUserLocation).toBeCalled();
+  await waitFor(() => {
+    expect(actions.setUserLocation).toBeCalledWith(expectedCoordinates);
+  });
+});
+
+it('updates rendered DisplayName if location changes and update button is clicked', async () => {
+  render(<LocationBias />);
+  const expectedLocationName = mockedStateVaDevice.location.locationBias.displayName;
+  const locationNameElement = screen.getByText(expectedLocationName);
+  expect(locationNameElement).toBeDefined();
+
+  mockAnswersState(mockedStateNyIP);
+  clickUpdateLocation();
+
+  await waitFor(() => {
+    expect(searchOperations.executeSearch).toBeCalled();
+  });
+
+  const newExpectedLocationName = mockedStateNyIP.location.locationBias.displayName;
+  const newLocationNameElement = screen.getByText(newExpectedLocationName);
+  expect(newLocationNameElement).toBeDefined();
+});
+
+it('renders correct attribution message, device', () => {
+  mockAnswersState(mockedStateVaDevice);
+  render(<LocationBias />);
+  const expectedMethodMessage = '(based on your device) -';
+  const basedOnMethodElement = screen.getByText(expectedMethodMessage);
+  expect(basedOnMethodElement).toBeDefined();
+});
+
+it('renders correct attribution message, IP', () => {
+  mockAnswersState(mockedStateNyIP);
+  render(<LocationBias />);
+  const expectedMethodMessage = '(based on your internet address) -';
+  const basedOnMethodElement = screen.getByText(expectedMethodMessage);
+  expect(basedOnMethodElement).toBeDefined();
+});
+
+it('renders nothing if there is no display name', () => {
+  mockAnswersState(mockedStateNoDisplayName);
+  render(<LocationBias />);
+
+  const updateLocationButton = screen.queryByText('Update your location');
+  expect(updateLocationButton).toBeNull();
```

Comment:
is there a way to check that nothing is rendered on the screen (not just that the update location button isn't there)?

Reply:
Thanks!  I was looking into this, and I also saw that some people rendered the element, changed it to a JSON and checked if it was null-- I thought that was a little weird, but this one makes sense.